pkg/types/validation/installconfig.go

```diff
@@ -824,6 +827,16 @@ func validateCompute(platform *types.Platform, control *types.MachinePool, pools
 			allErrs = append(allErrs, field.Invalid(poolFldPath.Child("fencing"), p.Fencing, "fencing is only valid for control plane"))
 		}
 	}
+
+	if control != nil {
+		controlReplicas = ptr.Deref(control.Replicas, 3)
+	}
+
+	// Validate that exactly 1 worker node is allowed only when a single control plane node is provisioned on platforms None or External
+	if computeReplicas == 1 && controlReplicas != 1 && !(platform.None != nil || platform.External != nil) {
```

Comment:
:thought_balloon:  The comment hints that 1 compute node is supported when configuring SNO **and** platform None/External. However, this check will allow: 1 compute + 1 control-plane regardless of platforms right ([pre-merge result](https://github.com/openshift/installer/pull/10042#issuecomment-3479141260))?

Maybe the comment is incorrect?